src/KnowledgeQuiz.tsx

```diff
@@ -42,35 +53,104 @@ const fieldsetStyle = css`
 export const KnowledgeQuizAtom = ({
     id,
     questions,
-}: QuizAtomType): JSX.Element => (
-    <form data-atom-id={id}>
-        {questions.map((question, idx) => (
-            <Question
-                key={question.id}
-                id={question.id}
-                number={idx + 1}
-                text={question.text}
-                imageUrl={question.imageUrl}
-                answers={question.answers}
-            />
-        ))}
-    </form>
-);
+    resultGroups,
+    sharingIcons,
+}: QuizAtomType): JSX.Element => {
+    const [selectedGlobalAnswers, setSelectedGlobalAnswers] = useState<{
+        [questionId: string]: AnswerType;
```

Comment:
we need to know if the selected answer is correct by checking its nested bool called `isCorrect`, so I thought it would be simpler to just map each question to thier selected answer